common/src/main/java/io/druid/metadata/PasswordProvider.java

```diff
@@ -34,5 +34,15 @@
 })
 public interface PasswordProvider
 {
+  // Used only for getting metadata storage connector password, present here for backwards compatibility
+  @Deprecated
   public String getPassword();
+
+  /**
+   * Implementations should provide a way to get password for the input key.
+   * For example {@link DefaultPasswordProvider} looks for a property named
+   * {@param key} prefixed with {@link DefaultPasswordProvider#PWD_PROPERTY_PREFIX}
+   * in the runtime properties and return the value.
+   */
+  public String getPassword(String key);
```

Comment:
I don't think this is a correct API for this, the configurations for the password provider can be supplied during the implementation construction, and having a "key" as a single object used to get passwords doesn't work in the general case. A simple example is an implementation that fetches an object from a blob storage, then uses a cloud KMS to decrypt that object. In this scenario you have both the blob storage path and the KMS coordinates that need configured, something that is not captured by a single `key` parameter.